ls1intum/Artemis
# `Programming exercises`: Automatically update PlantUML diagrams on theme changes

### Checklist
#### General


#### Client

### Motivation and Context
Fixes #7696

### Description
We will now trigger a markdown-update once the theme has changed. We also evict the plantuml caches since otherwise the old image would still get used.

### Steps for Testing

1. Open the detailed view of a programming exercise with at least one PlantUML-diagram
2. Switch the theme from light to dark and back
3. Observe that the diagram automatically adjusts to the changed theme

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress
#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage
changes covered
